frontend/src/metabase-lib/lib/queries/StructuredQuery.js

```diff
@@ -478,6 +480,20 @@ export default class StructuredQuery extends AtomicQuery {
     return this.table().segments.filter(sgmt => sgmt.is_active === true);
   }
 
+  /**
+   *  @returns @type {Segment}s that are currently applied to the question
+   */
+  segments() {
+    return this.filters()
+      .filter(f => isSegmentFilter(f))
+      .map(segmentFilter => {
+        // segment id is stored as the second part of the filter clause
+        // e.x. ["SEGMENT", 1]
+        const segmentId = segmentFilter[1];
+        return this.metadata().segments[segmentId];
```

Comment:
Could you add `segment` and `metric` methods to `frontend/src/metabase-lib/lib/metadata/Metadata.js` and use that?

```javascript
  segment(segmentId): ?Segment {
    return (segmentId != null && this.segments[segmentId]) || null;
  }
  metric(metricId): ?Metric {
    return (metricId != null && this.metrics[metricId]) || null;
  }
```